### PR TITLE
feat: disable start/restart if active version required

### LINKED ISSUE #10755: template update policies: explain why start is disabled
> We should give the end-user some indication in the UI that their workspace has no-update-restarts disabled. Idea: keep the restart button but show it as disabled with a hover explaining why.

### PR DESCRIPTION
**Pull request by sreya**
fixes https://github.com/coder/coder/issues/10755

Variants:

Template Required 
image

User opted-in 

image

Also Restart:

image

### CODE REVIEW

File: site/src/pages/WorkspacePage/WorkspaceActions/Buttons.tsx [hunk 1]
```diff
@@ -55,8 +58,14 @@ export const StartButton: FC<
...
     handleAction: (buildParameters?: WorkspaceBuildParameter[]) => void;
   }
-> = ({ handleAction, workspace, loading }) => {
-  return (
+> = ({
+  handleAction,
+  workspace,
+  loading,
+  disabled = false,
+  tooltipText = "",
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacePage/WorkspaceActions/Buttons.tsx:66]: I see a Golang developer here 😆

File: site/src/pages/WorkspacePage/WorkspaceActions/WorkspaceActions.tsx [hunk 2]
```diff
@@ -67,39 +68,70 @@ export const WorkspaceActions: FC<WorkspaceActionsProps> = ({
...
+    if (!disabled) {
+      return "";
+    }
+    if (workspace.template_require_active_version) {
+      return "This template requires automatic updates";
+    }
+    if (workspace.automatic_updates === "always") {
+      return "You have enabled automatic updates for this workspace";
+    }
+    return "";
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacePage/WorkspaceActions/WorkspaceActions.tsx:90]: I just found strange to have two places returning `""` 🤔

**Review by BrunoQuaresma — APPROVED** [review]
LGTM!

**Comment by BrunoQuaresma** [comment]
Question. Do we also want to disable the secondary action on start and restart? Both actions have a secondary action that can start/restart using ephemeral parameters. It is on the button with the "dropdown icon".

**Comment by sreya** [comment]
> Question. Do we also want to disable the secondary action on start and restart? Both actions have a secondary action that can start/restart using ephemeral parameters. It is on the button with the "dropdown icon". Screenshot 2023-11-21 at 09 21 40

Yeah we def do, thanks :heart:

**Pull request merged**